### ISSUE TITLE
Managing disk usage in Lustre

### ISSUE THREAD

**Issue by donaldgray** [dlcs/protagonist#280]
When serving images the Orchestrator is responsible for copying the asset from it's "slow/cheap" location (S3) to a "fask/expensive" location where the image-server can access it to serve the requests.

The Orchestrator has no knowledge of what underlying storage is being used (e.g. if it's a network drive or directly attached volume etc) or how full it is. It doesn't concern itself with disk cleanup, it expects there to always be room for images to be copied down. 

See [RFC-002](https://github.com/dlcs/protagonist/blob/develop/docs/rfcs/002-storage-and-orchestration.md) for further thinking on this issue.

## Current Process

We currently have a python [`echo_fs`](https://github.com/dlcs/echo-fs) process for maintaining orchestration disk space and removing old files. It uses a fairly basic LRU algorithm and will purge files that are older than X, when the disk has less than Y% free space remaining. At a high level, the process is:
* `echo_populate` periodically walks all files in a directory an adds path to a Redis SortedSet, scored by atime.
* `echo_scavenger` periodically runs; if disk space is less than Y% then it'll delete old files from SortedSet.

We also have a python [`scavenger`](https://github.com/digirati-co-uk/folder-scavenger) process for maintaining scratch disk space - this is temporary space used when ingesting images. It has a simpler implementation and shouldn't have a bearing on new work as services should clean up after themselves - only mentioning here for completeness.

## New Requirement

As part of an earlier [image-server spike](https://github.com/dlcs/cantaloupe-spike) we have decided to use AWS FSx for Lustre to store orchestrated images on - this will be available to both the orchestrator and image-servers.

This ticket is to investigate and implement something equivalent to `echo-fs` to ensure that the Lustre cluster doesn't fill up and older files are purged. This should use some sort of LRU algorithm to ensure that it's only old files that are purged. 

Exactly how this is done is yet to be decided e.g. Is there something smarter that we can do with Lustre to access file lists, (something exposed by the Lustre MDS that can be queried)?; do we use a similar approach to the existing process?, or something completely different. 

In short, it needs to:
* Monitor disk usage and ensure there's always room for orchestrated images.
* Do this in such a way that freshly orchestrated or used files are not deleted.
* Use an LRU algorithm to remove files from long tail.

Timeboxed spike (3 days) - output to will be an ADR on managing orchestration disk space from which implementation tasks can be created.

> The solution might require a change to Orchestrator, e.g. it may need to touch a file periodically when handling requests. If so, we can create a ticket for this.
